manifests/init.pp

```diff
@@ -0,0 +1,44 @@
+# Top level class for ClickHouse DBMS installation and management.
+#
+# @summary this class allows you to install ClickHouse DB's repo, client and server
+#
+# @param server [Boolean]
+#   Is clickhouse-server should be installed or not. It won't be installed by default, because some time you need the client only.
+# @param client [Boolean]
+#   Is clickhouse-client should be installed or not.
+# @param repo [Boolean]
+#   Is apt or yum repository should be managed. Set to `false` by default to not affect your own repositories policy.
+#
+# @example Simple use
+#   include clickhouse
+#
+# @example Install server and client
+#   class { 'clickhouse':
+#     server => true,
+#   }
+#
+# @example Install everything and manage repository
+#   class { 'clickhouse':
+#     server      => true,
+#     server      => true,
+#     manage_repo => true,
+#   }
+#
+class clickhouse (
```

Comment:
i am sorry to say so but this class doesn't make sense at all. also the dependencies are wrong here. `clickhouse::server` should require `clickhouse` and `clickhouse` should hold some global config. same with the other classes. `clickhouse::server` should require `clickhouse` only

Reply:
look for example how postgres does it. they have a dedicated `params` class like many other modules and then use those variables as default for the server class. https://github.com/puppetlabs/puppetlabs-postgresql/blob/master/manifests/server.pp

Reply:
We've decided to leave the module simple for now and place user and group to init. So everything convinced